### ISSUE TITLE
[Bug] Unable to serialize and then `json_encode` models

### ISSUE THREAD

**Issue by ellisonpatterson** [DirectoryTree/LdapRecord#429]
**Environment:**
 - LDAP Server Type: ActiveDirectory
 - LdapRecord-Laravel Major Version: 2.5.5
 - PHP Version: 8.0.13

**Describe the bug:**
If I make a model (extending Laravel Model or the LdapDirectory Model) and add the notifiable trait, I am unable to notify that user if I have the notification queue-able. You get this error:
```
[2022-04-08 18:23:00] local.ERROR: Unable to JSON encode payload. Error code: 5 {"exception":"[object] (Illuminate\\Queue\\InvalidPayloadException(code: 0): Unable to JSON encode payload. Error code: 5 at /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Queue/Queue.php:108)
[stacktrace]
#0 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Queue/RedisQueue.php(124): Illuminate\\Queue\\Queue->createPayload()
DirectoryTree/LdapRecord-Laravel#1 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(253): Illuminate\\Queue\\RedisQueue->push()
DirectoryTree/LdapRecord-Laravel#2 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(229): Illuminate\\Bus\\Dispatcher->pushCommandToQueue()
DirectoryTree/LdapRecord-Laravel#3 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(77): Illuminate\\Bus\\Dispatcher->dispatchToQueue()
DirectoryTree/LdapRecord-Laravel#4 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Notifications/NotificationSender.php(217): Illuminate\\Bus\\Dispatcher->dispatch()
DirectoryTree/LdapRecord-Laravel#5 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Notifications/NotificationSender.php(76): Illuminate\\Notifications\\NotificationSender->queueNotification()
DirectoryTree/LdapRecord-Laravel#6 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Notifications/ChannelManager.php(39): Illuminate\\Notifications\\NotificationSender->send()
DirectoryTree/LdapRecord-Laravel#7 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Notifications/RoutesNotifications.php(18): Illuminate\\Notifications\\ChannelManager->send()
DirectoryTree/LdapRecord-Laravel#8 /home/singlesignon/webapps/singlesignon/app/Jobs/PasswordExpiration.php(31): App\\Models\\User->notify()
DirectoryTree/LdapRecord-Laravel#9 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\\Jobs\\PasswordExpiration->handle()
DirectoryTree/LdapRecord-Laravel#10 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
DirectoryTree/LdapRecord-Laravel#11 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
DirectoryTree/LdapRecord-Laravel#12 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
DirectoryTree/LdapRecord-Laravel#13 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/Container.php(653): Illuminate\\Container\\BoundMethod::call()
DirectoryTree/LdapRecord-Laravel#14 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(128): Illuminate\\Container\\Container->call()
DirectoryTree/LdapRecord-Laravel#15 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Bus\\Dispatcher->Illuminate\\Bus\\{closure}()
DirectoryTree/LdapRecord-Laravel#16 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
DirectoryTree/LdapRecord-Laravel#17 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(132): Illuminate\\Pipeline\\Pipeline->then()
DirectoryTree/LdapRecord-Laravel#18 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(119): Illuminate\\Bus\\Dispatcher->dispatchNow()
DirectoryTree/LdapRecord-Laravel#19 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Queue\\CallQueuedHandler->Illuminate\\Queue\\{closure}()
DirectoryTree/LdapRecord-Laravel#20 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
DirectoryTree/LdapRecord-Laravel#21 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(121): Illuminate\\Pipeline\\Pipeline->then()
DirectoryTree/LdapRecord-Laravel#22 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(69): Illuminate\\Queue\\CallQueuedHandler->dispatchThroughMiddleware()
DirectoryTree/LdapRecord-Laravel#23 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(98): Illuminate\\Queue\\CallQueuedHandler->call()
DirectoryTree/LdapRecord-Laravel#24 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Queue/SyncQueue.php(43): Illuminate\\Queue\\Jobs\\Job->fire()
DirectoryTree/LdapRecord-Laravel#25 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(253): Illuminate\\Queue\\SyncQueue->push()
DirectoryTree/LdapRecord-Laravel#26 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(229): Illuminate\\Bus\\Dispatcher->pushCommandToQueue()
DirectoryTree/LdapRecord-Laravel#27 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(95): Illuminate\\Bus\\Dispatcher->dispatchToQueue()
DirectoryTree/LdapRecord-Laravel#28 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Foundation/Bus/Dispatchable.php(57): Illuminate\\Bus\\Dispatcher->dispatchSync()
DirectoryTree/LdapRecord-Laravel#29 /home/singlesignon/webapps/singlesignon/app/Console/Kernel.php(38): App\\Jobs\\PasswordExpiration::dispatchSync()
DirectoryTree/LdapRecord-Laravel#30 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(102): App\\Console\\Kernel->schedule()
DirectoryTree/LdapRecord-Laravel#31 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Support/helpers.php(302): Illuminate\\Foundation\\Console\\Kernel->Illuminate\\Foundation\\Console\\{closure}()
DirectoryTree/LdapRecord-Laravel#32 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(103): tap()
DirectoryTree/LdapRecord-Laravel#33 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/Container.php(873): Illuminate\\Foundation\\Console\\Kernel->Illuminate\\Foundation\\Console\\{closure}()
DirectoryTree/LdapRecord-Laravel#34 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\\Container\\Container->build()
DirectoryTree/LdapRecord-Laravel#35 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(855): Illuminate\\Container\\Container->resolve()
DirectoryTree/LdapRecord-Laravel#36 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\\Foundation\\Application->resolve()
DirectoryTree/LdapRecord-Laravel#37 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(840): Illuminate\\Container\\Container->make()
DirectoryTree/LdapRecord-Laravel#38 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(181): Illuminate\\Foundation\\Application->make()
DirectoryTree/LdapRecord-Laravel#39 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(124): Illuminate\\Container\\BoundMethod::addDependencyForCallParameter()
DirectoryTree/LdapRecord-Laravel#40 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Container\\BoundMethod::getMethodDependencies()
DirectoryTree/LdapRecord-Laravel#41 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
DirectoryTree/LdapRecord-Laravel#42 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
DirectoryTree/LdapRecord-Laravel#43 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
DirectoryTree/LdapRecord-Laravel#44 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Container/Container.php(653): Illuminate\\Container\\BoundMethod::call()
DirectoryTree/LdapRecord-Laravel#45 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Console/Command.php(171): Illuminate\\Container\\Container->call()
DirectoryTree/LdapRecord-Laravel#46 /home/singlesignon/webapps/singlesignon/vendor/symfony/console/Command/Command.php(291): Illuminate\\Console\\Command->execute()
DirectoryTree/LdapRecord-Laravel#47 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Console/Command.php(156): Symfony\\Component\\Console\\Command\\Command->run()
DirectoryTree/LdapRecord-Laravel#48 /home/singlesignon/webapps/singlesignon/vendor/symfony/console/Application.php(989): Illuminate\\Console\\Command->run()
DirectoryTree/LdapRecord-Laravel#49 /home/singlesignon/webapps/singlesignon/vendor/symfony/console/Application.php(299): Symfony\\Component\\Console\\Application->doRunCommand()
DirectoryTree/LdapRecord-Laravel#50 /home/singlesignon/webapps/singlesignon/vendor/symfony/console/Application.php(171): Symfony\\Component\\Console\\Application->doRun()
DirectoryTree/LdapRecord-Laravel#51 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Console/Application.php(102): Symfony\\Component\\Console\\Application->run()
DirectoryTree/LdapRecord-Laravel#52 /home/singlesignon/webapps/singlesignon/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\\Console\\Application->run()
DirectoryTree/LdapRecord-Laravel#53 /home/singlesignon/webapps/singlesignon/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
DirectoryTree/LdapRecord-Laravel#54 {main}
"} 
```
I found that it is some encoding issue and have to put this before I use notify:

```
$user->objectguid = $user->getConvertedGuid();
$user->objectsid = $user->getConvertedSid();
$user->syncOriginal();
```

Some type of UTF8 issue or something, but running those three lines allows the user model to be queues successfully without issue.

Could this be handled in the library potentially?

Thank you!

**Comment by stevebauman** [comment]
Moved this to the LdapRecord repository since this is a core issue 👍

**Comment by stevebauman** [comment]
Fixed in [v2.12.0](https://github.com/DirectoryTree/LdapRecord/releases/tag/v2.12.0) 🎉

**Comment by ellisonpatterson** [comment]
Thank you so much!
